lang/java/avro/src/test/java/org/apache/avro/io/TestValidatingIO.java

```diff
@@ -507,29 +496,27 @@ private static int skip(String msg, InputScanner cs, Decoder vi, boolean isArray
     throw new RuntimeException("Don't know how to skip");
   }
 
-  @Parameterized.Parameters
-  public static Collection<Object[]> data() {
-    return Arrays.asList(convertTo2dArray(encodings, skipLevels, testSchemas()));
+  public static Stream<Arguments> data() {
+    return convertTo2dArray(encodings, skipLevels, testSchemas());
   }
 
   private static Object[][] encodings = new Object[][] { { Encoding.BINARY }, { Encoding.BLOCKING_BINARY },
       { Encoding.JSON } };
 
   private static Object[][] skipLevels = new Object[][] { { -1 }, { 0 }, { 1 }, { 2 }, };
 
-  public static Object[][] convertTo2dArray(final Object[][]... values) {
-    ArrayList<Object[]> ret = new ArrayList<>();
+  public static Stream<Arguments> convertTo2dArray(final Object[][]... values) {
```

Comment:
Please rename (this method no longer creates an array), and remove the commented code (it's there in git, on the off chance anyone would want to look).